lib/SILOptimizer/Transforms/CSE.cpp

```diff
@@ -84,6 +84,18 @@ template <> struct DenseMapInfo<SimpleValue> {
 };
 } // end namespace llvm
 
+SILValue tryLookThroughOwnershipInsts(const Operand *op) {
+  auto opValue = op->get();
+  auto opOwnership = op->getOperandOwnership();
+
+  if (opOwnership == OperandOwnership::PointerEscape ||
+      opOwnership == OperandOwnership::BitwiseEscape ||
```

Comment:
I think we need a comment here because we don't normally consider BitwiseEscape to mean we can't optimize the lifetime. This looks like the usual check for "can we optimize this lifetime" but I think it's actually different. Here you're trying to catch all the cases that RAUW doesn't handle correctly, right? Can we just check exactly the same condition that RAUW checks and comment that they need to be in-sync?

Reply:
Unfortunately, there isn't any specific function that OSSA RAUW checks for. It doesn't bailout for none, unowned values, it replaces all uses with the replacement value without checking for lifetime.

Reply:
I chose to check for operand ownership here instead of just checking for none/unowned values because we should still be able to CSE instructions like` ref_element_addr` with a none result with another `ref_element_addr` looking through ownership instructions.